### DISCUSSION TITLE
Is there a way to automatically type the loader params?

### DISCUSSION THREAD

**Discussion by AsuraKev** [remix-run/react-router#10985] [General]
Hi there,

I am using the loader function every where but having to explicity type the loader param is tedious. I am wondering if I am doing it wrong or theres a way to automatically enforce the type definition automatically?
![image](https://github.com/remix-run/react-router/assets/52765572/4090ff91-6cf1-4680-bcac-8cbcd4d26c0e)

thanks

**Comment by kiliman** [comment]
I think you want the types `LoaderFunctionArgs` and `ActionFunctionArgs`

```ts
import type { ActionFunctionArgs, LoaderFunctionArgs } from "react-router-dom";

loader: ({request, params, context}: LoaderFunctionArgs) => { ... }
```
https://github.com/remix-run/react-router/blob/4475b56d2e18014acfb8ff5f6077ff27df98d7fe/packages/router/utils.ts#L135-L160

**Reply by aprilmintacpineda** [reply]
`LoaderFunctionArgs` is not generic though, how do we pass `params` type to it?

**Reply by nik-webdevelop** [reply]
@aprilmintacpineda +1 for this. Created separate discussion for this #11002
